File: src/event/impl/network/NetworkEvent.php
<?php

namespace pocketcloud\cloud\event\impl\network;

use pocketcloud\cloud\event\Event;
use pocketcloud\cloud\network\Network;

abstract class NetworkEvent extends Event {

    public function __construct(protected readonly Network $network) {}

    public function getNetwork(): Network {
        return $this->network;
    }
}

File: cloud-core/src/main/java/de/pocketcloud/cloud/event/impl/network/NetworkEvent.java
package de.pocketcloud.cloud.event.impl.network;

import de.pocketcloud.cloud.PocketCloud;
import de.pocketcloud.cloud.event.CloudEvent;
import de.pocketcloud.cloud.network.NetworkNettyServer;

public abstract class NetworkEvent extends CloudEvent {

    public NetworkNettyServer getNetwork() {
        return PocketCloud.instance().network();
    }
}